Doc/whatsnew/3.11.rst

```diff
@@ -167,6 +170,129 @@ default traceback. See :pep:`678` for more details. (Contributed by
 Irit Katriel in :issue:`45607`.)
 
 
+.. _new-feat-related-type-hints-311:
+
+New Features Related to Type Hints
+==================================
+
+This section covers major changes affecting :pep:`484` type hints and
+the :mod:`typing` module.
+
+PEP 646: Variadic generics
+--------------------------
+
+:pep:`484` introduced ``TypeVar``, enabling creation of generics parameterised
+with a single type. The new :data:`TypeVarTuple` enables parameterisation
+with an *arbitrary* number of types - that is, a *variadic* type variable,
+enabling *variadic* generics. This enables a wide variety of use cases.
+In particular, it allows the type of array-like structures
+in numerical computing libraries such as NumPy and TensorFlow to be
+parameterised with the array *shape*, enabling static type checkers
+to catch shape-related bugs in code that uses these libraries.
+
+See :pep:`646` for more details.
+
+(Contributed by Matthew Rahtz in :issue:`43224`, with contributions by
+Serhiy Storchaka and Jelle Zijlstra. PEP written by Mark Mendoza, Matthew
+Rahtz, Pradeep Kumar, and Vincent Siles.)
+
+PEP 655: Marking individual ``TypedDict`` items as required or non-required
+---------------------------------------------------------------------------
+
+:data:`typing.Required` and :data:`typing.NotRequired` provide a
+straightforward way to mark whether individual items in a
+:data:`TypedDict` must be present. Previously, this was only possible
+using inheritance.
+
+As before, fields are required by default, unless the ``total=False``
+parameter is set.
+For example, the following specifies a dictionary with one required and
+one non-required key::
+
+   class Movie(TypedDict):
+      title: str
+      year: NotRequired[int]
+
+   m1: Movie = {"title": "Black Panther", "year": 2018}  # ok
+   m2: Movie = {"title": "Star Wars"}  # ok (year is not required)
+   m3: Movie = {"year": 2022}  # error (missing required field title)
+
+The following definition is equivalent::
+
+   class Movie(TypedDict, total=False):
+      title: Required[str]
+      year: int
+
+See :pep:`655` for more details.
+
+(Contributed by David Foster and Jelle Zijlstra in :issue:`47087`. PEP
+written by David Foster.)
+
+PEP 673: ``Self`` type
+----------------------
+
+The new :data:`typing.Self` annotation provides a simple and intuitive
+way to annotate methods that returnan instance of their class. This
+behaves the same as the :data:`typing.TypeVar`-based approach specified
```

Comment:
Out of curiosity, does this render properly? Usually there needs to be an escaped space after a role.

If it does render properly, please ignore this comment.
```suggestion
behaves the same as the :data:`typing.TypeVar`\ -based approach specified
```

Reply:
<img width="773" alt="Screen Shot 2022-04-20 at 7 14 47 PM" src="https://user-images.githubusercontent.com/906600/164358047-b1c56678-cad9-44c6-9233-5d4bab379224.png">

Works fine